packages/compass-workspaces/src/components/workspaces.tsx

```diff
@@ -105,159 +102,117 @@ const CompassWorkspaces: React.FunctionComponent<CompassWorkspacesProps> = ({
-            title: tab.type,
-            iconGlyph: 'Logo',
-          } as const;
-        case 'My Queries':
-          return {
-            id: tab.id,
-            type: tab.type,
-            title: tab.type,
-            iconGlyph: 'CurlyBraces',
-          } as const;
-        case 'Data Modeling':
-          return {
-            id: tab.id,
-            type: tab.type,
-            title: tab.type,
-            iconGlyph: 'Diagram' as const,
-          };
-        case 'Shell': {
-          const connectionName =
-            getConnectionById(tab.connectionId)?.title || '';
-          const tooltip: Tooltip = [];
-          if (connectionName) {
-            tooltip.push(['mongosh', connectionName || '']);
-          }
-          return {
-            id: tab.id,
-            connectionName,
-            type: tab.type,
-            title: connectionName
-              ? `mongosh: ${connectionName}`
-              : 'MongoDB Shell',
-            tooltip,
-            iconGlyph: 'Shell',
-            tabTheme: getThemeOf(tab.connectionId),
-          } as const;
-        }
-        case 'Databases': {
-          const connectionName =
-            getConnectionById(tab.connectionId)?.title || '';
-          return {
-            id: tab.id,
-            connectionName,
-            type: tab.type,
-            title: connectionName,
-            tooltip: [['Connection', connectionName || '']] as Tooltip,
-            iconGlyph: 'Server',
-            tabTheme: getThemeOf(tab.connectionId),
-          } as const;
-        }
-        case 'Performance': {
-          const connectionName =
-            getConnectionById(tab.connectionId)?.title || '';
-          return {
-            id: tab.id,
-            connectionName,
-            type: tab.type,
-            title: `Performance: ${connectionName}`,
-            tooltip: [['Performance', connectionName || '']] as Tooltip,
-            iconGlyph: 'Gauge',
-            tabTheme: getThemeOf(tab.connectionId),
-          } as const;
-        }
-        case 'Collections': {
-          const connectionName =
-            getConnectionById(tab.connectionId)?.title || '';
+      const plugin = workspacePlugins.find((p) => p.name === tab.type);
+      if (!plugin) {
+        throw new Error(
+          `Content component for workspace "${tab.type}" is missing in context. Did you forget to set up WorkspacesProvider?`
+        );
+      }
+      const {
+        content: WorkspaceTabContent,
+        header: headerFn,
+        provider,
+      } = plugin;
+
+      const Provider = provider as any;
+
+      let connectionName: string | undefined;
+      if ('connectionId' in tab) {
+        connectionName = getConnectionById(tab.connectionId)?.title;
+      }
+
+      let isNonExistent: boolean | undefined;
+      if ('isNonExistent' in tab && tab.isNonExistent) {
+        if (tab.type === 'Collections') {
+          // TODO: Can/should we move this logic into the collections plugin title?
```

Comment:
I would expect it to be with the tab title rendering logic, yes

Reply:
Created a ticket for this to do as a follow up COMPASS-9456. Neither of the plugins have this in their store so it would take some more work and I'm thinking we could do that seperately.